frontend/app/src/store/blockchain/tokens.ts

```diff
@@ -96,7 +96,7 @@ export const useBlockchainTokensStore = defineStore('blockchain/tokens', () => {
   };
 
   watch(ethAddresses, async (curr, prev) => {
-    if (isEqual(curr, prev)) {
+    if (isEqual(curr, prev) || curr.length === 0) {
```

Comment:
It would be better to put the length check first.
If the length is zero then we don't need to check for equality.